### PR TITLE
Add regression test information to COMPILING.md

### PR DESCRIPTION
**Pull request by TGWDB**
This adds information on how to run regression tests using bnoth cmake and make to COMPILING.md.
 Also ensures COMPILING.md follows clang-format-7.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: REGRESSION_TESTS.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+simply running make as follows.
+```
+cd regression
+make
+```
+
+# CMAKE
+
+To build with release flags in cmake use
+`-DCMAKE_BUILD_TYPE=Release` as an argument to the cmake command.
```
Comment (hannes-steffenhagen-diffblue) [inline REGRESSION_TESTS.md]: This is unrelated to running the regression tests (it's already mentioned in `COMPILING.md` at any rate).

File: COMPILING.md [hunk 2]
```diff
@@ -255,6 +255,10 @@ require manual modification of build files.
    to generate IDE projects by supplying the `-G` flag.  Run `cmake -G` for a
    comprehensive list of supported back-ends.
 
+   Note that if you do not specify the `build` directory the current directory
+   will be the build target. This will modify many version controlled files
+   and be messy to fix.
```
Comment (hannes-steffenhagen-diffblue) [inline COMPILING.md]: I think I'll just add a change to cmake to make this impossible.

File: REGRESSION_TESTS.md (hunk 1 above)
Comment (TGWDB) [reply REGRESSION_TESTS.md]: Agree it is redundant. Removed.

File: COMPILING.md (hunk 2 above)
Comment (tautschnig) [inline COMPILING.md]: Rather than adding documentation I'd prefer one of the options discussed in https://stackoverflow.com/questions/1208681/with-cmake-how-would-you-disable-in-source-builds/10306476#10306476 to disallow this behaviour.

File: REGRESSION_TESTS.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
+This document assumes you have already been able to build CPROVER on
+your chosen architecture. For details on building CPROVER see
+COMPILING.md.
```
Comment (tautschnig) [inline REGRESSION_TESTS.md]: Could we please merge this documentation with doc/architectural/compilation-and-development.md rather than adding another file?

File: REGRESSION_TESTS.md (hunk 1 above)
Comment (tautschnig) [reply REGRESSION_TESTS.md]: That's already documented in COMPILING.md?!

File: COMPILING.md (hunk 2 above)
Comment (hannes-steffenhagen-diffblue) [reply COMPILING.md]: @tautschnig I added https://github.com/diffblue/cbmc/pull/5782 for this.

File: REGRESSION_TESTS.md (hunk 3 above)
Comment (TGWDB) [reply REGRESSION_TESTS.md]: Happy to merge this file with another one. I started a draft pull request to make sure I wasn't putting bad information in or adding things in the wrong place. If we all agree on doc/architectural/compilation-and-development.md as the right location I'll move everything there.

**Comment by hannes-steffenhagen-diffblue** [comment]
@tautschnig This CSmith check just failed: https://github.com/diffblue/cbmc/pull/5781/checks?check_run_id=1762881802

How do we interpret that?

File: COMPILING.md (hunk 2 above)
Comment (TGWDB) [reply COMPILING.md]: Removed as this is now superseded by PR #5782.

Comment (TGWDB) [reply COMPILING.md]: Handled by #5782

**Comment by tautschnig** [comment]
> @tautschnig This CSmith check just failed: https://github.com/diffblue/cbmc/pull/5781/checks?check_run_id=1762881802
> 
> How do we interpret that?

See #5780 - hoping for @kroening or @chrisr-diffblue to review this ASAP.

File: REGRESSION_TESTS.md (hunk 3 above)
Comment (tautschnig) [reply REGRESSION_TESTS.md]: Thank you for bringing up all these gaps in our documentation, and coming up with fixes for these! I'd just like to make sure we don't have the documentation in multiple places. I'm no longer able to tell what's actually needed for someone to get started, and where they might be looking for documentation. Is perhaps `CONTRIBUTING.md` (which we don't currently have) the canonical place for people to get started?

Comment (TGWDB) [reply REGRESSION_TESTS.md]: I'm happy to try and sort it all out. I'm using myself as a test subject for this (before I know my way around well), so hopefully do not create too many (draft) proposals to change things that are explained somewhere.

For (a possible) `CONTRIBUTING.md`, would this also include the various links and information on policies for style and committing?

Comment (tautschnig) [reply REGRESSION_TESTS.md]: If we had a `CONTRIBUTING.md`, perhaps it could serve as a collection of pointers towards documentation, pointing people to `COMPILING.md`, `CODING_STANDARD.md`, and the sections of the manual (such as `doc/architectural/compilation-and-development.md`).

Comment (peterschrammel) [reply REGRESSION_TESTS.md]: Btw, the documentation is published at http://cprover.diffblue.com (see link on top fo README)

Comment (TGWDB) [reply REGRESSION_TESTS.md]: What is our link preference here, should we link to http://cprover.diffblue.com/ or do we reference inside the repository since that page is generated from doc/architectural/front-page.md ?

Comment (tautschnig) [reply REGRESSION_TESTS.md]: `README.md` references http://cprover.diffblue.com/, so I'd go with that in top-level files (`README.md`, `COMPILING.md`, `CODING_STANDARD.md`, and possibly any files to be added at that level).

File: COMPILING.md [hunk 4]
```diff
@@ -482,3 +482,42 @@ successfully on Windows or macOS.
    The argument for the IPASIR parameter gives the build system the location for
    the IPASIR headers, which is needed for the cbmc includes of `ipasir.h`. The
    compiled binary will be placed in `cbmc/src/cbmc/cbmc`.
+This document assumes you have already been able to build CPROVER on
+your chosen architecture. For details on building CPROVER see
+COMPILING.md.
```
Comment (chrisr-diffblue) [inline COMPILING.md]: Since this content is _in_ `COMPILING.md` - maybe its better to say `see section BLAH` ?

File: COMPILING.md [hunk 5]
```diff
@@ -482,3 +482,42 @@ successfully on Windows or macOS.
...
+that will run all CORE tests that include `goto` in their
+name.
+
+## MAKE
+
+This can be done by changing to the regression directory and
+simply running make as follows.
+```
+cd regression
+make
```
Comment (chrisr-diffblue) [inline COMPILING.md]: I believe there's a slight difference between cmake/make in what tests actually run, which might be worth either clarifying, or adjusting the instructions to elide. That is around unit tests... I believe the `ctest` commands run _both_ regression _and_ unit tests, whereas I though the make command you show here _only_ runs the regression tests. My recommendation might be to tweak the section title to say "RUNNING REGRESSION AND UNIT TESTS", and then tweak the instructions for make to show how to run unit tests as well?

File: COMPILING.md (hunk 4 above)
Comment (TGWDB) [reply COMPILING.md]: Fixed

File: COMPILING.md (hunk 5 above)
Comment (TGWDB) [reply COMPILING.md]: I think this is clearer now.

**Review by chrisr-diffblue — APPROVED** [review]
Thanks, thats a great addition to the docs!
